php/blocks/class-blocks.php

```diff
@@ -0,0 +1,176 @@
+<?php
+/**
+ * Blocks
+ * 
+ * @package CoAuthors
+ */
+
+namespace CoAuthors;
+
+use WP_REST_Request;
+
+/**
+ * Blocks
+ */
+class Blocks {
+	/**
+	 * Construct
+	 */
+	public function __construct() {
+		add_action( 'init', array( __CLASS__, 'initialize_blocks' ) );
+	}
+
+	/**
+	 * Initialize Blocks
+	 */
+	public static function initialize_blocks() : void {
+
+		if ( ! apply_filters( 'coauthors_plus_support_blocks', true ) ) {
+			return;
+		}
+
+		add_action( 'render_block_context', array( __CLASS__, 'provide_author_archive_context' ), 10, 2 );
+		add_action( 'enqueue_block_editor_assets', array( __CLASS__, 'enqueue_store' ) );
+
+		/**
+		 * Templating functions used by many block render functions.
+		 */
+		require_once __DIR__ . '/templating/class-templating.php';
+
+		/**
+		 * Individual blocks.
+		 */
+		require_once __DIR__ . '/block-coauthors/class-block-coauthors.php';
+		Blocks\Block_CoAuthors::register_block();
+
+		require_once __DIR__ . '/block-coauthor-avatar/class-block-coauthor-avatar.php';
+		Blocks\Block_CoAuthor_Avatar::register_block();
+
+		require_once __DIR__ . '/block-coauthor-description/class-block-coauthor-description.php';
+		Blocks\Block_CoAuthor_Description::register_block();
+
+		require_once __DIR__ . '/block-coauthor-name/class-block-coauthor-name.php';
+		Blocks\Block_CoAuthor_Name::register_block();
+
+		require_once __DIR__ . '/block-coauthor-featured-image/class-block-coauthor-featured-image.php';
+		Blocks\Block_CoAuthor_Featured_Image::register_block();
+	}
+
+	/**
+	 * Provide Author Archive Context
+	 *
+	 * @param array $context, 
+	 * @param array $parsed_block
+	 * @return array
+	 */
+	public static function provide_author_archive_context( array $context, array $parsed_block ) : array {
```

Comment:
I'm still trying to get my head around the blocks.

When a CoAuthors Name is in a CoAuthors Block on a single post, then I get the right details.
If it's outside of the CoAuthors Block, then nothing appears.
Editor:

<img width="353" alt="Screenshot 2023-09-10 at 18 00 22" src="https://github.com/Automattic/Co-Authors-Plus/assets/88371/c7643c51-3271-418f-ad61-2a2f31343e71">

Front-end:
<img width="353" alt="Screenshot 2023-09-10 at 18 00 34" src="https://github.com/Automattic/Co-Authors-Plus/assets/88371/17eaff00-aa06-4a7e-8157-2b5127859afe">

If I amend the Author Archives template in FSE with the same CoAuthors Name, and CoAuthors Block with CoAuthors Name inside, then the back end is:

<img width="217" alt="Screenshot 2023-09-10 at 18 02 16" src="https://github.com/Automattic/Co-Authors-Plus/assets/88371/1effafdf-feeb-4dc3-9013-5340db941276">

Front-end:

<img width="293" alt="Screenshot 2023-09-10 at 18 03 28" src="https://github.com/Automattic/Co-Authors-Plus/assets/88371/9e6d46e4-f52e-4a88-b709-41904dcbde46">

If certain blocks are meant for only being used on author archive template, and other block(s) are meant for post / post templates, can we make that clear in the block name/descriptions please?

Reply:
> If certain blocks are meant for only being used on author archive template, and other block(s) are meant for post / post templates, can we make that clear in the block name/descriptions please?

I did play further with this on the author archive, and I can now see where a CoAuthors Block would work (inside the post inside the query loop) and a separate CoAuthor Name (I need to note the singular vs plural block naming better too) as the page title along with Description and Featured Image as well:

<img width="1907" alt="Screenshot 2023-09-10 at 18 42 47" src="https://github.com/Automattic/Co-Authors-Plus/assets/88371/62a0dcf8-6089-4e23-a587-f646506abeb8">

Reply:
I tried to address this by expanding the readme here: https://github.com/Automattic/Co-Authors-Plus/pull/997/commits/41967d8e78860fe7160b7b54ec300b0a10ab2c1c